fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/FedoraTypesUtils.java

```diff
@@ -389,13 +435,63 @@ public static void touchLdpMembershipResource(final Node node) {
     }
 
     /**
-     * Update the fedora:lastModified date of the node.
+     * Updates the LAST_MODIFIED_DATE and LAST_MODIFIED_BY properties to now, and the current user
+     * respectively.
      *
      * @param node The JCR node
+     *
      */
     public static void touch(final Node node) {
+        touch(node, null, null, null, null);
+    }
+
+    /**
+     * Updates the LAST_MODIFIED_DATE and LAST_MODIFIED_BY properties to the provided values.
+     *
+     * @param node The JCR node
+     * @param modified the modification date, or null if not explicitly set
+     * @param modifyingUser the userID who modified this resource or null if not explicitly set
+     *
+     */
+    public static void touch(final Node node, final Calendar modified, final String modifyingUser) {
+        touch(node, null, null, modified, modifyingUser);
+    }
+
+    /**
+     * Updates the LAST_MODIFIED_DATE, LAST_MODIFIED_BY, CREATED_DATE and CREATED_BY properties to the provided values.
+     *
+     * @param node The JCR node
+     * @param created the date the resource was created, or null if not explicitly set
+     * @param creatingUser the userID of created this resource or null if not explicitly set
+     * @param modified the modification date, or null if not explicitly set
+     * @param modifyingUser the userID who modified this resource or null if not explicitly set
+     *
+     */
+    public static void touch(final Node node, final Calendar created, final String creatingUser,
+                             final Calendar modified, final String modifyingUser) {
         try {
-            node.setProperty(FEDORA_LASTMODIFIED, getInstance(getTimeZone("UTC")));
+            if (created != null) {
+                node.setProperty(FEDORA_CREATED, created);
```

Comment:
Initially I thought to apply some constraints, but I've tended toward treating this feature as "giving the user some rope", if they want to hang themself, that's their business.  I can think of scenarios where even the most well-meaning constraints (like the one you propose) would frustrate a user.  

One example is a scenario where I want to mirror one repo in another, so I'd  have a camel route that wrote updates from one server to the next.  If, the system time wasn't right on one of the boxes, I wouldn't want the repo throwing out my updates.